cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -275,56 +280,127 @@ void accumulate_vertex_results(
+          auto offset = vertex - v_first;
+          return distances[offset] == (d - 1);
+        });
+      
+      // Create key_bucket_t from the frontier vertices
+      key_bucket_t<vertex_t, void, multi_gpu, true> vertex_list(handle);
+      vertex_list.insert(reusable_vertex_buffer.begin(), reusable_vertex_buffer.begin() + frontier_count);
+
+      // Compute deltas for frontier vertices
+      per_v_transform_reduce_outgoing_e(
+        handle,
+        graph_view,
+        vertex_list,
+        src_properties.view(),
+        dst_properties.view(),
+        cugraph::edge_dummy_property_t{}.view(),
+        [d] __device__(auto, auto, auto src_props, auto dst_props, auto) {
+          if ((thrust::get<0>(dst_props) == d) && (thrust::get<0>(src_props) == (d - 1))) {
+            auto sigma_v = static_cast<weight_t>(thrust::get<1>(src_props));
+            auto sigma_w = static_cast<weight_t>(thrust::get<1>(dst_props));
+            auto delta_w = thrust::get<2>(dst_props);
+            
+            return (sigma_v / sigma_w) * (1 + delta_w);
+          } else {
+            return weight_t{0};
+          }
+        },
+        weight_t{0},
+        reduce_op::plus<weight_t>{},
+        reusable_delta_buffer.begin(),
+        do_expensive_check);
+      
+      // Scatter deltas back to main deltas array
+      thrust::scatter(
+        handle.get_thrust_policy(),
+        reusable_delta_buffer.begin(),
+        reusable_delta_buffer.begin() + frontier_count,
+        reusable_vertex_buffer.begin(),
+        deltas.begin()
+      );
+      
+      // Update edge properties with current deltas
+      update_edge_src_property(
+        handle,
+        graph_view,
+        vertex_list.begin(),
+        vertex_list.end(),
+        thrust::make_zip_iterator(distances.begin(), sigmas.begin(), deltas.begin()),
+        src_properties.mutable_view());
+      update_edge_dst_property(
+        handle,
+        graph_view,
+        vertex_list.begin(),
+        vertex_list.end(),
+        thrust::make_zip_iterator(distances.begin(), sigmas.begin(), deltas.begin()),
+        dst_properties.mutable_view());
+
+      // Update centrality values for frontier vertices
+      thrust::gather(
+        handle.get_thrust_policy(),
+        reusable_vertex_buffer.begin(),
+        reusable_vertex_buffer.begin() + frontier_count,
+        centralities.begin(),
+        reusable_centrality_buffer.begin()
+      );
+      
+      thrust::transform(
+        handle.get_thrust_policy(),
+        reusable_centrality_buffer.begin(),
+        reusable_centrality_buffer.begin() + frontier_count,
+        reusable_delta_buffer.begin(),
+        reusable_centrality_buffer.begin(),
+        thrust::plus<weight_t>()
+      );
+      
+      thrust::scatter(
+        handle.get_thrust_policy(),
+        reusable_centrality_buffer.begin(),
+        reusable_centrality_buffer.begin() + frontier_count,
+        reusable_vertex_buffer.begin(),
+        centralities.begin()
```

Comment:
You can combine this to a single thrust call.

thrust::for_each(handle.get_thrust_policy(),
thrust::make_zip_iterator(vertex_list.begin(), resuable_delta_buffer.begin()),
thrust::make_zip_iterator(vertex_list.begin(), resuable_delta_buffer.begin()) + vertex_list.size(),
[...]__device__(auto pair) {
  centralities[thrust::get<0>(pair)] += thrust::get<1>(pair);
}